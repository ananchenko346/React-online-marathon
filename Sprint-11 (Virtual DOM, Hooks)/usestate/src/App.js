import React, {useState} from 'react';

import './App.css';

export default function App() {
  const [text, setText] = useState('React Marathon');

  const changeText = () => {
    setText(text.toLowerCase());
  }

  return <div onClick={changeText}>{text}</div>
}

