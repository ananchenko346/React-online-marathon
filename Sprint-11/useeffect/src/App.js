import {useState, useEffect} from 'react';

import './App.css';

export default function App() {

  const [state, setState] = useState(null);

  useEffect(() => {
    const data = localStorage.getItem('appData');
    setState(data);
  }, [])

  return (
    <div>
      React Marathon, appData: <input size="5" defaultValue={state}></input>
    </div>
  );
}
