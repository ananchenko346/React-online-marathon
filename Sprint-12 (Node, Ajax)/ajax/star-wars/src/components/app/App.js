import {Route, Routes} from 'react-router-dom';
import People from '../pages/people/People';
import Planets from '../pages/planets/Planets';
import Ships from '../pages/ships/Ships';
import '../../style/theme/bootstrap.min.css';

import './App.css';

const App = () => {
  
  return (
    <div className="app">
    <People/>
    </div>
  );
}

export default App;
