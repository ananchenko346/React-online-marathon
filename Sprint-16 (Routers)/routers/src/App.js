import React from 'react';
import {Route, Routes} from 'react-router-dom';

import Layout from './components/Layout';
import Subtask1 from './components/Subtask1';
import Subtask2 from './components/Subtask2';
import Subtask3 from './components/Subtask3';
import Subtask4 from './components/Subtask4';

import './App.css';

export default function App() {
  return (
    <div className="App">
      <Routes>
        <Route path="/" element={<Layout/>}>
          <Route path="subtask1/:id" element={<Subtask1/>}/>
          <Route path="subtask2" element={<Subtask2/>}/>
          <Route path="subtask3" element={<Subtask3/>}/>
          <Route path="subtask4" element={<Subtask4/>}/>
        </Route>
        <Route path="*" element={<Layout/>}/>
      </Routes>
    </div>
  );
}
