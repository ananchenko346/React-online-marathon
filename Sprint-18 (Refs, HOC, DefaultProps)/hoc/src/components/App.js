import {useState} from 'react';

import EntitiesList from './EntitiesList';
import EntityDetails from './EntityDetails';
import {fetchUsers, fetchUser} from '../api';

import './App.css';

function App() {
  const [userId, setUserId] = useState(1);
  const onUserClick = (id) => setUserId(id);

  return (
    <div>
      <EntitiesList
        onEntityDetailsClick={onUserClick}
        fetchMethod={fetchUsers}
        propsToDisplay={{name: 'Name', username: 'Username', email: 'Email'}}/>
      <EntityDetails
        fetchMethod={fetchUser}
        params={userId}
        propsToDisplay={{
          name: 'Name', username: 'Username', email: 'Email',
          address: 'Address', city: 'City', street: 'Street', suite: 'Apartments',
          phone: 'Phone'
        }}
      />
    </div>
  );
}

export default App;
