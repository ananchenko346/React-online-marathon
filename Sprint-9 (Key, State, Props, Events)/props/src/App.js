import React from 'react';

import First from './First';
import elements from './base';

import './App.css';

export default function App() {
  
  return (
    <div>
      Some data:
      <First elements={elements}/>
    </div>
  );
}
