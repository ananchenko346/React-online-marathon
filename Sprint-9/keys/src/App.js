import React from 'react';

import elements from './base';

import './App.css';


export default function App() {

  const element = elements.map(item => {
    const uniqueId = Math.floor(Math.random() * 100) + 1;
    return <li key={uniqueId}>{item}</li>;
  });

  return (
    <div>
      Some List:
      <ul>
        {element}
      </ul>
    </div>
  );
}
