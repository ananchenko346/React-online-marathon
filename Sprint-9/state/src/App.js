import React from 'react';
import {Component} from 'react';

import './App.css';

export default class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      appData: 'React Marathon'
    }
  }

  changeText = () => {
    if (this.state.appData === 'React Marathon') {
      this.setState({
        appData: this.state.appData.toLowerCase()
      })
    } else {
      this.setState({
        appData: 'React Marathon'
      })
    }
  }

  render() {
    const {appData} = this.state;
    return (
      <div onClick={this.changeText}>
        {appData}
      </div>
    )
  }
}
