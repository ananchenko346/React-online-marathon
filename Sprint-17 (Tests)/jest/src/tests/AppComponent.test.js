import renderer from 'react-test-renderer';

import App from '../components/App/index';

describe('review App', () => {
  it('render page', () => {
    const prop = renderer.create(<App />);
    expect(prop.toJSON()).toMatchSnapshot();
  });
});

