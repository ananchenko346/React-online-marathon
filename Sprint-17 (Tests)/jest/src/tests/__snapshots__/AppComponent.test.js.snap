// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`review App render page 1`] = `
<div
  className="App"
>
  <h1>
    Hello World!
  </h1>
  <div>
    <h3>
      GitHub User Info
    </h3>
    <ul />
  </div>
  <div>
    <h3>
      GitHub User Info
    </h3>
    <ul />
  </div>
</div>
`;
